lib/matplotlib/tests/test_axes.py

```diff
@@ -8290,3 +8290,18 @@ def test_extent_units():
     with pytest.raises(ValueError,
                        match="set_extent did not consume all of the kwargs"):
         im.set_extent([2, 12, date_first, date_last], clip=False)
+
+
+def test_repr_error_message():
+
+    def get_next_color():
+        return 'blue'  # pragma: no cover
+    import re
+    msg = re.escape(
+            "'c' argument must be a color, a sequence of colors"
+            ", or a sequence of numbers, not 'red\\n'"
+        )
```

Comment:
Ah, I missed the nested import; there's no need for one though, if you mark the string as raw. Also, I'd make the function name a bit more explicit.

```suggestion
def test_scatter_color_repr_error():
    def get_next_color():
        return 'blue'  # pragma: no cover

    msg = (
        r"'c' argument must be a color, a sequence of colors, "
        r"or a sequence of numbers, not 'red\\n'"
    )
```

Reply:
Updated both of those changes. Thank you! I was confused on that `re.escape`. This makes total sense.